orchagent/neighorch.cpp

```diff
@@ -516,11 +513,15 @@ bool NeighOrch::removeNeighbor(NeighborEntry neighborEntry)
         gCrmOrch->decCrmResUsedCounter(CrmResourceType::CRM_IPV6_NEIGHBOR);
     }
 
-    NeighborUpdate update = { neighborEntry, MacAddress(), false };
-    notify(SUBJECT_TYPE_NEIGH_CHANGE, static_cast<void *>(&update));
+    SWSS_LOG_NOTICE("Removed neighbor %s on %s",
+            m_syncdNeighbors[neighborEntry].to_string().c_str(), alias.c_str());
 
     m_syncdNeighbors.erase(neighborEntry);
     m_intfsOrch->decreaseRouterIntfsRefCount(alias);
```

Comment:
why do we have this code here? is this an existing bug?

Reply:
So if the update is sent before the `m_syncdNeighbors.erase`, when the observer wants to query the NeighOrch to get the information, the deprecated entry is still in the m_syncdNeighbors data structure, which will cause issues. That is why I mvoe the update code position.